temporallib/client/client.py

```diff
@@ -50,29 +51,32 @@ def _init_runtime_with_prometheus(port: int) -> Runtime:
 
 class Client:
     """
-    A class which wraps the :class:`temporalio.client.Client` class
+    A class which wraps the :class:`temporalio.client.Client` class with reconnect logic.
     """
-    
+
     _is_stop_token_refresh = False
 
     @classmethod
-    def __del__(self):
-        self._is_stop_token_refresh = True
+    def __del__(cls):
+        cls._is_stop_token_refresh = True
 
     @classmethod
-    async def update_rpc_metadata_loop(self, client_opt, rpc_metadata):
+    async def reconnect_loop(cls):
         """
-        Periodically update the rpc_metadata headers
+        Reconnects to the Temporal server periodically when the token expires.
         """
-        while not self._is_stop_token_refresh:
-            # By default, refresh every 55 minutes. This is because Google OAuth
-            # tokens expire after 60 minutes.
-            await asyncio.sleep(3300)
-            auth_header_provider = AuthHeaderProvider(client_opt.auth)
-            rpc_metadata.update(auth_header_provider.get_headers())
+        while not cls._is_stop_token_refresh:
```

Comment:
question: This means that after you set this to `False`, it can send another request 55 minutes later. Are we OK with that?

Reply:
The point is that this flag will remain False while this connection is initialized, meaning we re-initialize the connection with a new token every 55 minutes.